Blockzilla/URLBar.swift

```diff
@@ -664,15 +664,22 @@ class URLBar: UIView {
     fileprivate func setTextToURL() {
         var fullUrl: String? = nil
         var truncatedURL: String? = nil
-
+        var displayText: String? = nil
+        
         if let url = url {
             // Strip the username/password to prevent domain spoofing.
             var components = URLComponents(url: url, resolvingAgainstBaseURL: false)
             components?.user = nil
             components?.password = nil
             fullUrl = components?.url?.absoluteString
             truncatedURL = components?.host
-            urlText.text = isEditing ? fullUrl : truncatedURL
+
+            if let stackValue = SearchHistoryUtils.pullSearchFromStack(), !stackValue.hasPrefix("http://") && !stackValue.hasPrefix("https://") {
+                displayText = stackValue
+            } else {
+                displayText = truncatedURL
+            }
+            urlText.text = isEditing ? fullUrl : displayText
```

Comment:
Looks like this still isn't working as expected. Rather than checking for http/https (which isn't always a given), you might want to check if it can be a URL. You can verify that your solution works by comparing it to the urls currently displayed when you run v7.0-dev. 

Example: twitter.com (when using autocomplete)
Expected:
<img width="373" alt="image" src="https://user-images.githubusercontent.com/20130899/44108422-353d03c4-9faf-11e8-9b85-ca3d673d5968.png">

Result:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/20130899/44108453-4ff77fbe-9faf-11e8-8329-6a9b9b51e4a1.png">